src/grub.cfg

```diff
@@ -57,4 +57,16 @@ if [ -f "/ignition.firstboot" ]; then
     set ignition_firstboot="ignition.firstboot ${ignition_network_kcmdline}"
 fi
 
+# Read ignition platform id from /ignition_platform_id
+set ignition_platform_id=""
+if [ -f "/ignition_platform_id" ]; then
+    set ignition.platform.id=''
+
+    # source in the `ignition_platform_id` file which could override the
+    # above $ignition.platform.id with ignition platform id.
+    source "/ignition_platform_id"
+
+    set ignition_platform_id="${ignition.platform.id}"
```

Comment:
I don't quite follow what's going on here. Shouldn't the sourced file be something like:
```
set ignition_platform_id=qemu
```
Or did you have something else in mind?

Reply:
From my understanding, I'm trying to source the `ignition.platform.id=qemu` and introduce `ignition.platform.id` variable inside the sourced file and assign it to `ignition_platform_id`. Then in the `create_disk.sh` apply the set `ignition_platform_id` kernel argument.

Reply:
What about making the kcmdline option `ignition.platform.id=$ignition_platform_id` and keeping this part simpler?